chain-api/src/types/DexPositionData.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import BigNumber from "bignumber.js";
+import { Exclude, Type } from "class-transformer";
+import { IsInt, IsNotEmpty, IsString, Max, Min, ValidateNested } from "class-validator";
+import { JSONSchema } from "class-validator-jsonschema";
+
+import { ChainKey, ConflictError, requirePosititve } from "../utils";
+import { BigNumberIsNotNegative, BigNumberProperty, EnumProperty, IsLessThan } from "../validators";
+import { ChainObject } from "./ChainObject";
+import { DexFeePercentageTypes } from "./DexDtos";
+import { TickData } from "./TickData";
+import { TokenClassKey } from "./TokenClass";
+
+@JSONSchema({
+  description:
+    `Represents a liquidity position in a decentralized exchange (DEX) pool.` +
+    `Each position is associated with a unique NFT and defined by tick boundaries, liquidity amount, and fee tracking information.`
+})
+export class DexPositionData extends ChainObject {
+  @Exclude()
+  static INDEX_KEY = "GCDPD"; //GalaChain Dex Position Data
+
+  @ChainKey({ position: 0 })
+  @IsNotEmpty()
+  @IsString()
+  poolHash: string;
+
+  @ChainKey({ position: 1 })
+  @IsInt()
+  @Max(TickData.MAX_TICK)
+  tickUpper: number;
+
+  @ChainKey({ position: 2 })
+  @IsInt()
+  @Min(TickData.MIN_TICK)
+  @IsLessThan("tickUpper")
+  tickLower: number;
+
+  @ChainKey({ position: 4 })
```

Comment:
```suggestion
  @ChainKey({ position: 2 })
  @IsInt()
  @Min(TickData.MIN_TICK)
  @IsLessThan("tickUpper")
  tickLower: number;

  @ChainKey({ position: 3 })
```

Keep the `@ChainKey` positions sequential

Reply:
Fixed the skipped chainkey position index.